### PR TITLE
Configure Fastlane for distributing to AppTester

### PR DESCRIPTION
**Pull request by superus8r**
Relates OR Closes #0000

## Description
- configure Fastlane to distribute the example app to the firebase app tester
- add a Circleci step to run Fastlane distribute
- add an icon badge for fastlane debug builds

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -92,6 +92,19 @@ jobs:
...
+    steps:
+      - checkout
+      - restore_gradle_cache
+      - run: ./gradlew createEmptyConfigFile
+      - run:
+          name: Prepare fastlane
+          command: |
+            ./gradlew generateAppDistKey
+            sudo bundle update
+      - run: bundle exec fastlane distDev
```
Comment (timo-reymann) [inline .circleci/config.yml:106]: suggestion: Add name for run

File: .circleci/config.yml [hunk 2]
```diff
@@ -107,3 +120,4 @@ workflows:
             branches:
               ignore:
                 - main
+      - distribute-dev
```
Comment (timo-reymann) [inline .circleci/config.yml:123]: question: Is it intended to distribute a dev version on all branches?

File: app/build.gradle.kts [hunk 3]
```diff
@@ -65,6 +65,22 @@ tasks.register("createEmptyConfigFile") {
     }
 }
 
+tasks.register("generateAppDistKey") {
+    doLast {
+        val jsonFileName = "app-dist-key.json"
+        val fileContent = System.getenv("GOOGLE_APP_DIST_FASTLANE_SERVICE_ACCOUNT")
```
Comment (timo-reymann) [inline app/build.gradle.kts:71]: praise: Nice use of env var

File: fastlane/report.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (timo-reymann) [inline fastlane/report.xml:1]: suggestion: Remove test execution file from git

**Review by timo-reymann — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (superus8r) [reply .circleci/config.yml:123]: No, it's not required for the main branch 🙂

**Review by timo-reymann — APPROVED** [review]

**Review by timo-reymann — APPROVED** [review]

**Review by timo-reymann — APPROVED** [review]

**Review by timo-reymann — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 5]
```diff
@@ -97,11 +97,21 @@ jobs:
     steps:
       - checkout
 #      - restore_gradle_cache
+      - run: ./gradlew clean
```
[the remark this answers is not in the document]
Comment (superus8r) [reply .circleci/config.yml]: it is added ✅

**Pull request merged**